src/mlmc/sample_storage_hdf.py

```diff
@@ -86,6 +86,11 @@ def save_result_format(self, result_format: List[QuantitySpec], res_dtype):
         :param result_format: List[QuantitySpec]
         :return: None
         """
+        try:
```

Comment:
The result format must be set before storing the values, we should probably provide it in the constructor.